### PR TITLE
Support initial choice on CardSelectionView

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-8780

## Changes
- Add selection predicate to CardSelectionView

### CODE REVIEW

File: src/foam/u2/view/CardSelectionView.js [hunk 1]
```diff
@@ -97,6 +100,11 @@ foam.CLASS({
                 cardSelectViewConfig.data$ = valueSimpSlot;
                 cardSelectViewConfig.data  = valueSimpSlot.get();
 
+                // Set choice selection based on predicate
+                if ( self.selectionPredicate && self.selectionPredicate.f(choice) ) {
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/CardSelectionView.js]: Wouldnt this override the existing self.data if someone populates that property directly?

File: src/foam/u2/view/CardSelectionView.js [hunk 2]
```diff
@@ -58,6 +58,11 @@ foam.CLASS({
       name: 'choiceView',
       value: { class: 'foam.u2.view.CardSelectView' }
     },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.mlang.predicate.Predicate',
+      name: 'selectionPredicate'
```
Comment (kgrgreer) [inline src/foam/u2/view/CardSelectionView.js]: Please add documentation:

Comment (kgrgreer) [reply src/foam/u2/view/CardSelectionView.js]: If this defaulted to FALSE, then you would need the null check below.

File: src/foam/u2/view/CardSelectionView.js (hunk 1 above)
Comment (lchanmann) [reply src/foam/u2/view/CardSelectionView.js]: Fixed in https://github.com/kgrgreer/foam3/pull/2410/commits/bd25ca5a53f5acb87d957d75b09fae6c4d58e194.

**Review by sarthak-marwaha — APPROVED** [review]

File: src/foam/u2/view/CardSelectionView.js (hunk 2 above)
Comment (lchanmann) [reply src/foam/u2/view/CardSelectionView.js]: it's undefined by default the optional chaining operator should be able take care of it.

**Pull request merged**